### ISSUE TITLE
[3.2.x] Non-Identity Models Don't Save Snapshot on Creation/Save

### ISSUE THREAD

**Issue by virgofx** [phalcon/cphalcon#13166]
When saving a new model that does NOT have an identity field (e.g. using a primary UUID instead) the snapshot does not get saved which results in errors when retrieving the updated fields.

```php
$model = new NonIdentityModel();
$model->setKeys = 1;
$model->save();
$model->getChangedFields(); <--- Exception thrown because snapshot !== array
```

The `Model::_doLowInsert()` needs to move the snapshot save portion out of the identity-only block as snapshots are independent of whether the model has an autoincrement field or not.

![image](https://user-images.githubusercontent.com/739719/32972025-ecdd14f4-cba4-11e7-9c49-21a54b30eecf.png)

**Comment by Jurigag** [comment]
Oh, good catch :+1:

**Comment by Jurigag** [comment]
This can be closed as it is fixed already in 3.3.x

**Comment by virgofx** [comment]
Closed /fixed in 3.3.x
